framework/src/modules/chain/components/storage/entities/block.js

```diff
@@ -183,6 +183,22 @@ class ChainBlock extends BlockEntity {
 			{}
 		);
 	}
+
+	/**
+	 * Get the matching highest block for the providen array of block ids
+	 * @param {Array} blockIds - An array of block ids
+	 * @return {Promise<BasicBlock>}
+	 */
+	async getMatchingHighestBlock(blockIds) {
```

Comment:
Isn't this too specific to the logic? 
Probably we can just put this logic in the `blocks.js`

Reply:
I don't have a strong opinion regarding this, for me it is correct in both ways. @nazarhussain discussed this with me and can give his input.

Reply:
@nazarhussain what's your opinion on this?
I feel like so far most of the functions on entities are generic, and we should keep it that way

Reply:
After looking at the implementation I agree to @shuse2. Since it don't have any custom SQL or complex logic behind, so can be used only in the implementation as well. 

```
entities.Block.get(
 			{
 				id_in: blockIds,
 			},
 			{ sort: 'height:desc', limit: 1 }
 		)
```

Reply:
@shuse2 @nazarhussain logic has been moved back to its place.Done